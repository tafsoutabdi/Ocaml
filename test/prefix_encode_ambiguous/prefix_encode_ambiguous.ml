open Tokenization.PrefixTree.Tokenizer
open Utils.PrefixTree
open Utils.Std

(**
encodage du vocabulaire

nada 9
ada 5
c 1
ca 6
can 7
canal 3

**)

let prefix_tree = Node {id = None; successors = [
  ('n', Node {id = None; successors = [
    ('a', Node {id = None; successors = [
      ('d', Node {id = None; successors =[
        ('a', Node {id = Some 9; successors = []})]})]})]});
  ('a', Node {id = None; successors = [
    ('d', Node {id = None; successors = [
      ('a', Node {id = Some 5; successors = []})]})]});
  ('c',Node {id = Some 1; successors = [
    ('a', Node {id = Some 6; successors =[
      ('n', Node {id = Some 7; successors = [
        ('a', Node {id = None; successors = [
          ('l', Node {id = Some 3; successors = []})]})]})]})]})]}

let print_context prefix_tree s = 
  Format.printf "%s@.s = \"%s\"@.encode_aux prefix_tree s = "
    (string_of_prefix_tree prefix_tree) s


let () = 
  let s = "canada" in
  print_context prefix_tree s;
  print_id_list (encode_aux prefix_tree s);
  Format.printf "@."