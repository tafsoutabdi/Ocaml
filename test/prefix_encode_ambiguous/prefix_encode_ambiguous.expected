let prefix_tree =
let node3 = {id=Some(5); successors=[]} in
let node2 = {id=None; successors=[('a', Node node3)]} in
let node1 = {id=None; successors=[('d', Node node2)]} in
let node8 = {id=Some(3); successors=[]} in
let node7 = {id=None; successors=[('l', Node node8)]} in
let node6 = {id=Some(7); successors=[('a', Node node7)]} in
let node5 = {id=Some(6); successors=[('n', Node node6)]} in
let node4 = {id=Some(1); successors=[('a', Node node5)]} in
let node12 = {id=Some(9); successors=[]} in
let node11 = {id=None; successors=[('a', Node node12)]} in
let node10 = {id=None; successors=[('d', Node node11)]} in
let node9 = {id=None; successors=[('a', Node node10)]} in
Node {id=None; successors=[('a', Node node1); ('c', Node node4); ('n', Node node9)]}

s = "canada"
encode_aux prefix_tree s = [7; 5]
